ophyd/areadetector/plugins_orchestrators.py

```diff
@@ -0,0 +1,43 @@
+from .plugins import ProcessPlugin
+
+
+class ProcessPluginOrch(ProcessPlugin):
+    def configure_average(self, num_to_avg=1, **kwargs):
+        """
+        num_to_avg: int
+
+        Configures ProcessPlugin to output
+        and average of the last num_to_avg images
+        """
+
+        # Most intuitive configuration is
+        # set by default. Can be overriden by user with kwargs.
+        self.enable_background.put(0)
+        self.enable_flat_field.put(0)
+        self.enable_offset_scale.put(0)
+
+        self.enable_filter.put(1)
+        self.enable.put(1)
+        self.filter_type.put(1)  # Average
+
+        self.auto_reset_filter.put(num_to_avg)
+        self.filter_callbacks.put(1)  # Array N only
+        self.num_filter.put(num_to_avg)
+
+        # Overriding with kwargs part
+        properties = {
+            kwargs.get("enable_background"): self.enable_background,
```

Comment:
This is going to make the keys the values in kwargs (many of which will be `None`).  I think what you want is something like

```
for k, v in kwargs:
    getattr(self, k).put(v)
```

Reply:
I tried something similar to:

```
for k in kwargs.values():
    getattr(self.cam, k).put(kwargs[k])
```
    
but for some reason the operation completed without errors but also without changing the PV values.

I tried testing again now but for some reason, cloning my own fork, installing it with pip and testing it with the same SimDetector IOC container image i used before throws an NotImplementedError every time i try to execute a put operation... what am i doing wrong?

```
>>> from ophyd.areadetector.detectors  import SimDetector
>>> s = SimDetector(prefix="SIM:", name="SIM")
>>> s.cam.acquire.put(1)
Traceback (most recent call last):
  File "/home/marco/.local/lib/python3.10/site-packages/ophyd/device.py", line 325, in __get__
    return instance._signals[self.attr]
KeyError: 'acquire'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/marco/.local/lib/python3.10/site-packages/ophyd/device.py", line 327, in __get__
    return instance._instantiate_component(self.attr)
  File "/home/marco/.local/lib/python3.10/site-packages/ophyd/device.py", line 1352, in _instantiate_component
    self._signals[attr] = cpt.create_component(self)
  File "/home/marco/.local/lib/python3.10/site-packages/ophyd/device.py", line 266, in create_component
    cpt_inst = self.cls(pv_name, parent=instance, **kwargs)
  File "/home/marco/.local/lib/python3.10/site-packages/ophyd/areadetector/base.py", line 36, in __init__
    super().__init__(prefix + "_RBV", write_pv=prefix, **kwargs)
  File "/home/marco/.local/lib/python3.10/site-packages/ophyd/signal.py", line 1740, in __init__
    super().__init__(
  File "/home/marco/.local/lib/python3.10/site-packages/ophyd/signal.py", line 971, in __init__
    self._read_pv = self.cl.get_pv(
  File "/home/marco/.local/lib/python3.10/site-packages/ophyd/_dummy_shim.py", line 51, in get_pv
    raise NotImplementedError
NotImplementedError
```

shouldnt this be implemented in the base class?